src/rollup/config.ts

```diff
@@ -150,9 +151,13 @@ export const getRollupConfig = (nitro: Nitro): RollupConfig => {
   // WASM import support
   if (nitro.options.experimental.wasm) {
     const options = {
-      ...(nitro.options.experimental.wasm as RollupWasmOptions),
+      ...(nitro.options.experimental.wasm as WasmOptions),
     };
-    rollupConfig.plugins.push(wasmPlugin(options));
+    if (options.directImport) {
+      rollupConfig.plugins.push(wasmImport());
+    } else {
+      rollupConfig.plugins.push(wasmPlugin(options));
```

Comment:
We might merge wasmPlugin into the wasm import to keep config (little bit) simplified.